arangod/VocBase/Methods/Collections.cpp

```diff
@@ -419,21 +488,47 @@ Result Collections::create(TRI_vocbase_t& vocbase, OperationOptions const& optio
   TRI_ASSERT(infoSlice.length() >= 1);
   TRI_ASSERT(infoSlice.length() == infos.size());
   collections.reserve(infoSlice.length());
+
   try {
     if (ServerState::instance()->isCoordinator()) {
-      collections =
-          ClusterMethods::createCollectionOnCoordinator(vocbase, infoSlice, false,
-                                                        createWaitsForSyncReplication,
-                                                        enforceReplicationFactor,
-                                                        isNewDatabase, colToDistributeShardsLike);
+      // Here we do have a cluster setup. In that case, we will create many collections in one go (batch-wise).
+      collections = ClusterMethods::createCollectionsOnCoordinator(
+              vocbase, infoSlice, false, createWaitsForSyncReplication,
+              enforceReplicationFactor, isNewDatabase, colToDistributeShardsLike);
 
       if (collections.empty()) {
         for (auto const& info : infos) {
           events::CreateCollection(vocbase.name(), info.name, TRI_ERROR_INTERNAL);
         }
         return Result(TRI_ERROR_INTERNAL, "createCollectionsOnCoordinator");
       }
+    } else if (isSingleServerEnterpriseCollection) {
+#ifdef USE_ENTERPRISE
+      // todo: add comment to describe that section && implement
+      TRI_ASSERT(ServerState::instance()->isSingleServer());
+      TRI_ASSERT(infoSlice.isArray());
+
+      auto res =
+          enterprise::CollectionValidatorEE::createLogicalCollections(infoSlice, collections,
```

Comment:
Why does a Validator create collections?
I think conceptually that should not be the validators job.

Further, why is this different from the classical SingleServer Variant?
Can't we just incorporate the specialities into the normal single server branch?

In addition I am very confused by the `isSingleServerEnterpriseCollection` (which implies I only have one) but this API actually handles a list.
It this just a flag that we allow the API to be used?
Is there a reason to not use the API by default in Enterprise variant?

Reply:
Yes there is a reason as on the cluster side we're calling: 
`createCollectionsOnCoordinator`

(which is internally also calling that exact same `createLogicalCollection` method, but directly afterwards is handling agency e.g.

Reply:
We can rename the class itself. e.g. CollectionCreator rather then Validator.

Reply:
renamed && resolved.